prereise/gather/demanddata/nrel_efs/map_states.py

```diff
@@ -2,29 +2,56 @@
 
 import pandas as pd
 from powersimdata.input.grid import Grid
-from powersimdata.network.usa_tamu.constants.zones import abv2state, id2abv, id2timezone
+from powersimdata.network.usa_tamu.constants.zones import (
+    abv2state,
+    id2abv,
+    id2timezone,
+    interconnect2id,
+)
 
 
-def decompose_demand_profile_by_state_to_loadzone(df, save=None):
+def decompose_demand_profile_by_state_to_loadzone(
+    df, profile_type, regions=None, save=None
+):
     """Transforms the sectoral demand data so that it is separated by load zone rather
     than by state.
 
     :param pandas.DataFrame df: DataFrame of the sectoral demand data, where the rows
         are time steps (in local time) and the columns are the states. This input is
         intended to be the output of :py:func:`combine_efs_demand` or the components
         that are output from :py:func:`partition_flexibility_by_sector`.
+    :param str profile_type: A str that identifies the type of profile that is provided.
+        Can be one of *'demand'* or *'demand_flexibility'*.
+    :param set/list regions: The combination of interconnection names and state
```

Comment:
Maybe we can put `iterable` here?

Reply:
I'm not sure I follow where you are suggesting that be added. Are you suggesting that replace `set/list`?

Reply:
Yeah, I remember previously we did replace `list/set/tuple` by `iterable`.

Reply:
Okay, will do!

Reply:
Done.